src/sentry/replays/post_process.py

```diff
@@ -138,3 +136,47 @@ def _archived_row(replay_id: str, project_id: int) -> dict[str, Any]:
         "started_at": None,
         "is_archived": True,
     }
+
+
+CLICK_FIELD_MAP = {
+    "click_alt": "click.alt",
+    "click_aria_label": "click.label",
+    "click_classes": "click.classes",
+    "click_id": "click.id",
+    "click_role": "click.role",
+    "click_tag": "click.tag",
+    "click_testid": "click.testid",
+    "click_text": "click.text",
+    "click_title": "click.title",
+}
+
+
+def extract_click_fields(item: MutableMapping[str, Any]) -> list[dict[str, Any]]:
+    """
+    pops all of the click fields from the item and returns a list of the individual clicks as objects
+    """
+    click_fields = [
+        "click_alt",
+        "click_aria_label",
+        "click_classes",
+        "click_id",
+        "click_role",
+        "click_tag",
+        "click_testid",
+        "click_text",
+        "click_title",
+    ]
+    click_dict = {}
+    for click_field in click_fields:
+        click_val = item.pop(click_field, [None])
+        # if there is at least one one element, the list will be filled empty strings for the non-click segments
+        # so if there is at least one value, return a list of the truthy values.
+        # if not, return a list with a single None value
+        # also map the clickhouse field values to their query names
+        if click_val:
+            click_dict[CLICK_FIELD_MAP[click_field]] = [element for element in click_val if element]
+        else:
+            click_dict[CLICK_FIELD_MAP[click_field]] = click_val
+
+    list_of_dicts = [dict(zip(click_dict, col)) for col in zip_longest(*click_dict.values())]
```

Comment:
This took me a minute to understand.  I get it now but I'll forget again.  I'm thinking we should rename `col` to `row` or `columns` and use `click_dict.keys()` to explicitly annotate what we're iterating over in the zip function.